skorch/tests/callbacks/test_training.py

```diff
@@ -547,32 +548,42 @@ def sink(x):
         assert expected_msg in msg
 
         # Recompute validation loss and store it together with module weights
-        y_proba = net.predict_proba(val_dataset)
-        es_weights = deepcopy(net.module_.state_dict())
+        y_proba = net1.predict_proba(val_dataset)
+        es_weights = deepcopy(net1.module_.state_dict())
         es_loss = log_loss(y_val, y_proba)
 
         # Retrain same classifier without ES, using the best epochs number
-        net = net_clf_cls(
+        net2 = net_clf_cls(
             classifier_module,
             max_epochs=early_stopping_cb.best_epoch_,
             train_split=predefined_split(val_dataset),
         )
         torch.manual_seed(seed)
-        net.fit(tr_dataset, y=None)
+        net2.fit(tr_dataset, y=None)
 
         # Check that weights obtained match
         assert all(
             torch.equal(wi, wj)
             for wi, wj in zip(
-                net.module_.state_dict().values(),
+                net2.module_.state_dict().values(),
                 es_weights.values()
             )
         )
 
         # Check validation loss obtained match
-        y_proba_2 = net.predict_proba(val_dataset)
+        y_proba_2 = net2.predict_proba(val_dataset)
         assert es_loss == log_loss(y_val, y_proba_2)
 
+        # Check best_model_weights_ is transformed into None when pickling
+        with tempfile.NamedTemporaryFile() as tmp_file:
```

Comment:
You could use `pickle.dumps` and `pickle.loads` so that you don't need to create an actual file.

Reply:
Oh yes, that's way simpler ! :)